eng/pipelines/client.yml

```diff
@@ -15,3 +15,4 @@ jobs:
   parameters:
     ServiceDirectory: $(ServiceDirectory)
     VersioningProperties: /p:OfficialBuildId=$(OfficialBuildId) /p:PreRelaseVersionLabel=$(PreReleaseVersionLabel) /p:DotNetFinalVersionKind=$(DotNetFinalVersionKind)
+    BuildConfiguration: $(BuildConfiguration)
```

Comment:
To help avoid issues and duplication across all our many pipelines. Can we default this directly in the template? I'm imagining something like:
```
variables:
  ${{ if eq(variables['System.TeamProject'], 'internal') }}
     Configuration: Release
```

Since it defaults to Debug everywhere else I think this should take care of it for us.

Reply:
@mitchdenny  I'd be interested in getting your opinion on this approach as well.

Reply:
Could there be a situation where we still want a debug build from this pipeline?